src/cellRenderers/renderCheckbox.tsx

```diff
@@ -1,88 +1,44 @@
 import { css } from '@linaria/core';
-import clsx from 'clsx';
 
 import type { RenderCheckboxProps } from '../types';
 
-const checkboxLabel = css`
-  @layer rdg.CheckboxLabel {
-    cursor: pointer;
+const checkboxContainer = css`
+  @layer rdg.checkboxContainer {
     display: flex;
     align-items: center;
     justify-content: center;
     position: absolute;
     inset: 0;
-    margin-inline-end: 1px; /* align checkbox in row group cell */
   }
 `;
 
-const checkboxLabelClassname = `rdg-checkbox-label ${checkboxLabel}`;
-
-const checkboxInput = css`
-  @layer rdg.CheckboxInput {
-    position: absolute;
-    inline-size: 20px;
-    block-size: 20px;
-    z-index: 1;
-    cursor: inherit;
-    opacity: 0;
-  }
-`;
-
-const checkboxInputClassname = `rdg-checkbox-input ${checkboxInput}`;
-
 const checkbox = css`
-  @layer rdg.CheckboxIcon {
-    content: '';
+  @layer rdg.CheckboxInput {
     inline-size: 20px;
     block-size: 20px;
-    border: 2px solid var(--rdg-border-color);
-    background-color: var(--rdg-background-color);
-
-    .${checkboxInput}:checked + & {
-      background-color: var(--rdg-checkbox-color);
-      outline: 4px solid var(--rdg-background-color);
-      outline-offset: -6px;
-    }
+    accent-color: var(--rdg-checkbox-color);
 
-    .${checkboxInput}:focus + & {
-      border-color: var(--rdg-checkbox-focus-color);
+    &:focus-visible {
+      outline: 2px solid var(--rdg-checkbox-focus-color);
+      outline-offset: -3px;
     }
-  }
-`;
-
-const checkboxClassname = `rdg-checkbox ${checkbox}`;
-
-const checkboxLabelDisabled = css`
-  @layer rdg.CheckboxLabel {
-    cursor: default;
 
-    .${checkbox} {
-      border-color: var(--rdg-checkbox-disabled-border-color);
-      background-color: var(--rdg-checkbox-disabled-background-color);
+    &:enabled {
+      cursor: pointer;
     }
   }
 `;
 
-const checkboxLabelDisabledClassname = `rdg-checkbox-label-disabled ${checkboxLabelDisabled}`;
+const checkboxClassname = `rdg-checkbox-input ${checkbox}`;
 
 export function renderCheckbox({ onChange, ...props }: RenderCheckboxProps) {
   function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
     onChange(e.target.checked, (e.nativeEvent as MouseEvent).shiftKey);
   }
 
   return (
-    <label
-      className={clsx(checkboxLabelClassname, {
-        [checkboxLabelDisabledClassname]: props.disabled
-      })}
-    >
-      <input
-        type="checkbox"
-        {...props}
-        className={checkboxInputClassname}
-        onChange={handleChange}
-      />
-      <div className={checkboxClassname} />
-    </label>
+    <div className={checkboxContainer}>
```

Comment:
I wonder if we should make `checkboxContainer` a column class name, instead of rendering an extra absolutely-positioned div

Reply:
We probably can but it requires updating many columns. May be this is okay for now